File: dataset/output/GEMINI_gemini-1.5-pro-001/task165/Task165_GEMINI_gemini-1.5-pro-001.java
import java.util.HashMap;
class Solution {
    public boolean checkFlush(String[] table, String[] hand) {
      HashMap<Character, Integer> suits = new HashMap<>();
      suits.put('S', 0);
      suits.put('H', 0);
      suits.put('D', 0);
      suits.put('C', 0);
      for (String card : table) {
        suits.put(card.charAt(card.length()-1), suits.get(card.charAt(card.length()-1)) + 1);
      }
      for (String card : hand) {
        suits.put(card.charAt(card.length()-1), suits.get(card.charAt(card.length()-1)) + 1);
      }
      for (int val : suits.values()) {
        if (val >= 5) {
          return true;
        }
      }
      return false;
    }
}

File: dataset/output/GEMINI_gemini-1.5-pro-001/Task165_GEMINI_gemini-1.5-pro-001.cpp
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>

using namespace std;

bool checkFlush(vector<string> table, vector<string> hand) {
  unordered_map<char, int> suits;
  suits['S'] = 0;
  suits['H'] = 0;
  suits['D'] = 0;
  suits['C'] = 0;
  for (string card : table) {
    suits[card.back()]++;
  }
  for (string card : hand) {
    suits[card.back()]++;
  }
  for (auto it = suits.begin(); it != suits.end(); it++) {
    if (it->second >= 5) {
      return true;
    }
  }
  return false;
}